janusgraph-benchmark/src/main/java/org/janusgraph/MgmtOlapJobBenchmark.java

```diff
@@ -0,0 +1,127 @@
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This benchmark evaluates performance of OLAP jobs that
+ * can be run via the ManagementSystem interface, including:
+ * 1) REINDEX
+ * 2) REMOVE_INDEX
+ *
+ * @author Boxuan Li (liboxuan@connect.hku.hk)
+ */
+@BenchmarkMode(Mode.AverageTime)
+@Fork(1)
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class MgmtOlapJobBenchmark {
+    @Param("10000")
+    int size;
+
+    JanusGraph graph;
+
+    public WriteConfiguration getConfiguration() {
+        ModifiableConfiguration config = GraphDatabaseConfiguration.buildGraphConfiguration();
+        config.set(GraphDatabaseConfiguration.STORAGE_BACKEND, "inmemory");
+        return config.getConfiguration();
+    }
+
+    @Setup(Level.Iteration)
+    public void setUp() throws Exception {
+        graph = JanusGraphFactory.open(getConfiguration());
+
+        JanusGraphManagement mgmt = graph.openManagement();
+        PropertyKey name = mgmt.makePropertyKey("name").dataType(String.class).cardinality(Cardinality.SINGLE).make();
+        mgmt.buildIndex("nameIndex", Vertex.class).addKey(name).buildCompositeIndex();
+        mgmt.commit();
+        ManagementSystem.awaitGraphIndexStatus(graph, "nameIndex").call();
+
+        for (int j = 0; j < size; j++) {
+            graph.addVertex("name", "value" + j, "alias", "value" + j);
+        }
+        graph.tx().commit();
+
+        mgmt = graph.openManagement();
+        mgmt.buildIndex("aliasIndex", Vertex.class).addKey(mgmt.getPropertyKey("alias")).buildCompositeIndex();
+        mgmt.commit();
+        ManagementSystem.awaitGraphIndexStatus(graph, "aliasIndex").call();
+
+        mgmt = graph.openManagement();
+        mgmt.updateIndex(mgmt.getGraphIndex("nameIndex"), SchemaAction.DISABLE_INDEX).get();
+        mgmt.commit();
+        ManagementSystem.awaitGraphIndexStatus(graph, "nameIndex").call();
+    }
+
+    @Benchmark
+    public void runReindex(Blackhole blackhole) throws ExecutionException, InterruptedException {
+        JanusGraphManagement mgmt = graph.openManagement();
+        blackhole.consume(mgmt.updateIndex(mgmt.getGraphIndex("aliasIndex"), SchemaAction.REINDEX).get());
+        mgmt.commit();
+    }
+
+    @Benchmark
+    public void runRemoveIndex(Blackhole blackhole) throws ExecutionException, InterruptedException {
+        JanusGraphManagement mgmt = graph.openManagement();
+        blackhole.consume(mgmt.updateIndex(mgmt.getGraphIndex("nameIndex"), SchemaAction.REMOVE_INDEX).get());
+        mgmt.commit();
+    }
+
+
+    @TearDown(Level.Iteration)
+    public void tearDown() {
+        graph.close();
+    }
+
+    public static void main(String[] args) throws RunnerException {
```

Comment:
Do we need the main?

Reply:
Not needed for CI, but it would be handy to run this particular benchmark locally with the main function.